tests/api_fastapi/core_api/routes/public/test_dag_report.py

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from unittest.mock import patch
+
+import pytest
+import re2 as re
+
+from tests_common.test_utils.config import conf_vars
+from tests_common.test_utils.db import clear_db_dags
+
+pytestmark = pytest.mark.db_test
+
+
+class TestDagReportEndpoint:
+    @pytest.fixture(autouse=True)
+    def setup(self) -> None:
+        self.clear_db()
+
+    def teardown_method(self) -> None:
+        self.clear_db()
+
+    def clear_db(self):
+        clear_db_dags()
+
+    @pytest.mark.parametrize(
+        "subdir,include_example,expected_total_entries",
+        [
+            pytest.param("/root/airflow/dags", True, 58, id="dag_path_with_example"),
+            pytest.param("/root/airflow/dags", False, 0, id="dag_path_without_example"),
+        ],
+    )
+    def test_should_response_200(self, test_client, subdir, include_example, expected_total_entries):
+        with conf_vars({("core", "load_examples"): str(include_example)}):
+            params = {"subdir": subdir} if subdir else {}
```

Comment:
`subdir` is always defined.
```suggestion
            params = {"subdir": subdir}
```